From 4f9dde567eab4e3afe56bda71891a41b5a7ca2cd Mon Sep 17 00:00:00 2001
From: Roy Li <rongqing.li@windriver.com>
Date: Thu, 27 Feb 2014 14:37:05 +0800
Subject: [PATCH] Give resolv.conf the net_conf_t security label

Upstream-Status: Pending

Since resolv.conf becomes a link from /etc to /var/run, and the type_transition
rule {type_transition dhcpc_t var_run_t : file dhcpc_var_run_t} can not give
resolv.conf the correct label. so add a type_transition rule to fix it.

Signed-off-by: Roy Li <rongqing.li@windriver.com>
---
 policy/modules/system/sysnetwork.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/sysnetwork.te b/policy/modules/system/sysnetwork.te
index 3fd3912..3257d75 100644
--- a/policy/modules/system/sysnetwork.te
+++ b/policy/modules/system/sysnetwork.te
@@ -74,6 +74,8 @@ manage_files_pattern(dhcpc_t, dhcpc_var_run_t, dhcpc_var_run_t)
 create_dirs_pattern(dhcpc_t, dhcpc_var_run_t, dhcpc_var_run_t)
 files_pid_filetrans(dhcpc_t, dhcpc_var_run_t, { file dir })
 
+filetrans_pattern(dhcpc_t, var_run_t, net_conf_t, file, "resolv.conf.dhclient")
+filetrans_pattern(dhcpc_t, var_run_t, net_conf_t, file, "resolv.conf.dhclient6")
 # Allow read/write to /etc/resolv.conf and /etc/ntp.conf. Note that any files
 # in /etc created by dhcpcd will be labelled net_conf_t.
 sysnet_manage_config(dhcpc_t)
-- 
2.7.4

